x-pack/plugins/cross_cluster_replication/public/app/store/actions/auto_follow_pattern.js

```diff
@@ -135,9 +131,10 @@ export const deleteAutoFollowPattern = (id) => (
       }
 
       // If we've just deleted a pattern we were looking at, we need to close the panel.
-      const detailPanelAutoFollowPatternName = getDetailPanelAutoFollowPatternName(getState());
-      if (detailPanelAutoFollowPatternName && response.itemsDeleted.includes(detailPanelAutoFollowPatternName)) {
-        dispatch(closeAutoFollowPatternDetailPanel());
+      const ids = arrify(id);
+      const autoFollowPatternId = getSelectedAutoFollowPatternId(getState());
+      if (ids.some(_id => autoFollowPatternId === _id)) {
```

Comment:
The original code anticipated that the deletion request could fail for the selected auto-follow pattern. It looks like the new code doesn't anticipate that case and will close regardless of the request failing or succeeding. Is this correct? If so, then I think the first case makes more sense from the user's perspective. What do you think?

Reply:
This is inside the `onSuccess` handler so will only be executed if the deletion succeeded.